src/Map.ts

```diff
@@ -752,6 +762,37 @@ export class Map extends maplibregl.Map {
     });
 
     this.telemetry = new Telemetry(this);
+
+    // Set default value for defaultControls
+    options = {
+      defaultControls: true,
+      ...options,
+    };
+
+    // Check for potential control configuration issues
+    this.checkControlsConfiguration(options);
+  }
+
+  private checkControlsConfiguration(options: MapOptions): void {
+    if (options.defaultControls === false) {
+      const controlOptions = [
+        { name: "navigationControl", value: options.navigationControl },
+        { name: "geolocateControl", value: options.geolocateControl },
+        { name: "terrainControl", value: options.terrainControl },
+        { name: "projectionControl", value: options.projectionControl },
+        { name: "scaleControl", value: options.scaleControl },
+        { name: "fullscreenControl", value: options.fullscreenControl },
+      ];
+
+      controlOptions.forEach(({ name, value }) => {
+        if (value !== undefined && value !== false) {
```

Comment:
Given that the keys for the controls in `options` are the same as the string value in `name` would it not be cleaner to just iterate over an array of strings, or are the keys likely to differ from the name at some point?

```ts
if (options.defaultControls === false) {
  const controlOptions = [
    "navigationControl",
    "geolocateControl",
    "terrainControl",
    "projectionControl",
    "scaleControl",
    "fullscreenControl",
  ];

  controlOptions.forEach((controlKey) => {
    const control = options[controlKey];
    if (control !== undefined && control !== false) {
      console.warn('Hi imma warn you now')
    }
  });
}
```

Reply:
Actually I'm betting typescript might complain about that?

Reply:
Yes, you may need to create a record, but prefer to make it the way I'm doing.